### PR TITLE
Split board drawing and logic and draw gem sprites

### PR DESCRIPTION
**Pull request by sharkwouter**
This PR became a little bit bigger than planned. It also includes a change in the size of the gems and screen to make things a bit more visible.

### CODE REVIEW

File: src/Gem.hpp [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-    SDL_Texture *texture;
-    SDL_Rect src;
-    int x;
-    int y;
-public:
-    Gem(int dst_x, int dst_y, SDL_Texture *texture, SDL_Rect &src);
-
-    void update();
-    void draw(SDL_Renderer *renderer);
+enum Gem {
```
Comment (protheory8) [inline src/Gem.hpp]: Make that `enum class` instead.
```suggestion
enum class Gem {
```

File: src/Gem.hpp [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
+        RED,
+        ORANGE,
```
Comment (protheory8) [inline src/Gem.hpp]: Those have 8 spaces instead of 4.

File: src/constants.hpp [hunk 3]
```diff
@@ -1,9 +1,9 @@
 #ifndef CONSTANTS_HPP
 #define CONSTANTS_HPP
 
-inline constexpr int SCREEN_WIDTH = 800;
-inline constexpr int SCREEN_HEIGHT = 600;
+inline constexpr int SCREEN_WIDTH = 769;
+inline constexpr int SCREEN_HEIGHT = 673;
 
-const int GEM_SIZE = 32;
+const int GEM_SIZE = 96;
```
Comment (protheory8) [inline src/constants.hpp]: You might need to use `inline constexpr` instead? I don't remember to be honest, does it compile?

**Review by protheory8 — COMMENTED** [review]
Pretty good, just need to change those things.

**Comment by sharkwouter** [comment]
Thanks for reviewing. I've fixed the things you addressed

**Review by protheory8 — APPROVED** [review]
LGTM, add newline to the end of `src/BoardManager.hpp` file please though.

**Pull request merged**